core/src/impl/Kokkos_ViewCtor.hpp

```diff
@@ -273,6 +260,32 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   }
 };
 
+template <typename... P>
+auto add_properties(const ViewCtorProp<P...> &view_ctor_prop) {
+  return view_ctor_prop;
+}
+
+template <typename... P, typename Property, typename... Properties>
+auto add_properties(const ViewCtorProp<P...> &view_ctor_prop,
+                    const Property &property,
+                    const Properties &... properties) {
+  if constexpr ((is_execution_space<Property>::value &&
+                 !ViewCtorProp<P...>::has_execution_space) ||
+                (is_memory_space<Property>::value &&
+                 !ViewCtorProp<P...>::has_memory_space) ||
+                (is_view_label<Property>::value &&
+                 !ViewCtorProp<P...>::has_label) ||
+                (std::is_same_v<Property, WithoutInitializing_t> &&
+                 ViewCtorProp<P...>::initialize)) {
+    using NewViewCtorProp = ViewCtorProp<P..., Property>;
+    NewViewCtorProp new_view_ctor_prop(view_ctor_prop);
+    static_cast<ViewCtorProp<void, Property> &>(new_view_ctor_prop).value =
+        property;
+    return add_properties(new_view_ctor_prop, properties...);
+  } else
+    return add_properties(view_ctor_prop, properties...);
```

Comment:
Why is silently ignoring the property the right thing to do in the else-clause?

Reply:
Perhaps the function should be called "add default properties"

Reply:
Though that only makes sense for the already present case, not the unrecognized argument type case.

Reply:
Again, the code that is to be replaced by this function oftentimes expresses "add some property if it doesn't exist already" and that's the reason for ignoring properties in this branch. I'm open to changing the name of the function, of course.